src/main/java/eiteam/esteemedinnovation/boiler/TileEntityBoiler.java

```diff
@@ -195,29 +205,33 @@ public void safeUpdate() {
         }
         */
 
-        ItemStack stackInInput = getStackInSlot(1);
-        if (FluidHelper.itemStackIsWaterContainer(stackInInput)) {
-            ItemStack drainedItemStack = FluidHelper.fillTankFromItem(stackInInput, myTank, true);
-            setInventorySlotContents(1, drainedItemStack);
+        ItemStack stackInInput = inventory.getStackInSlot(1);
+        if (!stackInInput.isEmpty()) {
+            if (FluidHelper.itemStackIsWaterContainer(stackInInput)) {
+                ItemStack drainedItemStack = FluidHelper.fillTankFromItem(stackInInput, myTank, true);
+                inventory.setStackInSlot(1, drainedItemStack);
+            }
         }
 
         boolean isBurnTimeGreaterThanZero = burnTime > 0;
         if (burnTime > 0) {
             //maxThisTick = Math.min(burnTime, 10);
             burnTime -= 1; //maxThisTick
         }
+        if (burnTime < 0) {
```

Comment:
Why is this added?

Reply:
There was an issue where the burn time was somehow getting negative. This resolves that issue.

Reply:
It gets set to -1 on line 219. To be honest I don't quite understand how the burn time stuff works.

Reply:
That's a -= not = -

Reply:
Oh you're right. I misread that.